streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -1127,21 +1125,20 @@ private InternalTopicConfig createChangelogTopicConfig(final StateStoreFactory f
     }
 
     public synchronized Pattern earliestResetTopicsPattern() {
-        final List<String> topics = maybeDecorateInternalSourceTopics(earliestResetTopics);
-        final Pattern earliestPattern = buildPatternForOffsetResetTopics(topics, earliestResetPatterns);
-
-        ensureNoRegexOverlap(earliestPattern, latestResetPatterns, latestResetTopics);
-
-        return earliestPattern;
+        return resetTopicsPattern(earliestResetTopics, earliestResetPatterns, latestResetPatterns, latestResetTopics);
     }
 
     public synchronized Pattern latestResetTopicsPattern() {
-        final List<String> topics = maybeDecorateInternalSourceTopics(latestResetTopics);
-        final Pattern latestPattern = buildPatternForOffsetResetTopics(topics, latestResetPatterns);
+        return resetTopicsPattern(latestResetTopics, latestResetPatterns, earliestResetPatterns, earliestResetTopics);
+    }
+
+    private Pattern resetTopicsPattern(Set<String> resetTopics, Set<Pattern> resetPatterns, Set<Pattern> latestResetPatterns, Set<String> latestResetTopics) {
```

Comment:
nit: add `final` and single-parameter per line formatting

improve parameter names for `latestResetPatterns` and `latestResetTopics` (we pass in "earliest" here, too). Maybe just call the `otherResetPatterns` and `otherResetTopics`?

Reply:
`otherResetPatterns` and `otherResetTopics` look good to me

Reply:
done with changes